sites/svelte.dev/src/lib/server/examples/index.js

```diff
@@ -46,17 +48,17 @@ export function get_examples_data(base = CONTENT_BASE_PATHS.EXAMPLES) {
 
 			// Get title for
 			const example_title = JSON.parse(
-				fs.readFileSync(`${example_base_dir}/meta.json`, 'utf-8')
+				await readFile(`${example_base_dir}/meta.json`, 'utf-8')
 			).title;
 
 			const files = [];
-			for (const file of fs
-				.readdirSync(example_base_dir)
-				.filter((file) => !file.endsWith('meta.json'))) {
+			for (const file of (await readdir(example_base_dir)).filter(
+				(file) => !file.endsWith('meta.json')
+			)) {
 				files.push({
 					name: file,
 					type: file.split('.').at(-1),
-					content: fs.readFileSync(`${example_base_dir}/${file}`, 'utf-8')
+					content: await readFile(`${example_base_dir}/${file}`, 'utf-8')
```

Comment:
Does making this things noticably faster when going full async on the function? I fear that it will result in too many open async connections resulting in the opposite.

Reply:
I dunno, I just saw the need to dynamically import `fs` so edge workers wouldn't scream at me. Then I thought, that if the function is async, might as well use the async fs functions, cuz they're non blocking.

I didn't notice any reduction in speed, because the build times always range from 2 min to 3 minutes, so its not deterministic. 

I don't have any particular opinion on it. If you want I can revert back

Reply:
Why would edge workers not scream at you if you import them dynamically? If the module isn't available it doesn't matter at what time your import it. Also, this only runs during build if I'm not mistaken and you have access to all node modules there

Reply:
I think its a bug with kit or the adapter. If I import `get_examples_list` and `get_example`, which dont even access filesystem, I keep getting errors that `fs` is not supported and all. Even though `get_examples_data`, which is the one relying on fs doesn't even get imported here.

Thats why the need to dynamically import in the first place. If the bug was gone, we can go back to static imports et al

Reply:
I think the adapters try to do their special magic before properly tree shaking the final stuff, leading to this issue

Reply:
Probably doesn't hit that import because it only runs during build, but the static imports are kept for whatever side-effect reasons